File: C Language/Swap_Alternate_Elements_In_LinkedList.c
//Remove Duplicates from Sorted Singly Linked List

#include<stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node * next;
    struct node * prev;
};

struct node *head = NULL;

void linkedListTraversal(struct node* ptr)
{
    ptr = head;
    while(ptr != NULL)
    {
        printf("%d ", ptr -> data);
        ptr = ptr -> next;
    }
}

void createList(int data)
{
    struct node *ptr = (struct node*)malloc(sizeof(struct node));
    ptr -> data = data;
    ptr -> next = head;	
    head = ptr;
}

void SwapAlternateElement(struct node* ptr)
{
    
    ptr = head;
    int x;
    if(ptr == NULL)
    {
        return;
    }
    while(ptr != NULL && ptr -> next != NULL)
    {
        x = ptr -> data;
        ptr -> data = ptr -> next -> data;
        ptr -> next -> data = x;
        ptr = ptr -> next -> next;
    }
}


int main()
{
   
    createList(19);
    createList(70);
    createList(43);
    createList(6);
    createList(28);
    createList(8);
    createList(1);
    printf("Linked List before swapping\n");
    linkedListTraversal(head);
    
    SwapAlternateElement(head);
    printf("\n\nLinked List after swapping\n");
    linkedListTraversal(head);
    return 0;
}

File: Python/Swap_Alternate_Elements_In_LinkedList.py
class Node:
	def __init__(self, data):
		self.data = data
		self.next = None
		self.prev = None

class LinkedList:
	def __init__(self):
		self.head = None
			
	def push(self, data):
		newNode = Node(data)
		newNode.next = self.head
		self.head = newNode
		
	def printList(self, ptr):
		ptr = self.head
		if ptr is None:
			print('List has no element')
			return
		else:
			temp = ptr
			while(temp):
				print(temp.data, end=' ')
				temp = temp.next
				
	def SwapAlternateElements(self):
			ptr = self.head
			while ptr is not None and ptr.next is not None:
				x = ptr.data
				ptr.data = ptr.next.data
				ptr.next.data = x
				ptr = ptr.next.next
				
llist = LinkedList()

llist.push(9)
llist.push(24)
llist.push(30)
llist.push(15)
llist.push(28)
llist.push(12)
llist.push(1)

print("Linked List before swapping")
llist.printList(llist.head)

print("\n\nLinked List after swapping")
llist.SwapAlternateElements()
llist.printList(llist.head)